org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/ProjectsManager.java

```diff
@@ -302,6 +305,26 @@ public void fileChanged(String uriString, CHANGE_TYPE changeType) {
 		}
 	}
 
+	private IResource refreshRootFolders(String uriString) {
```

Comment:
I think automatic detection is probably not always safe (might break Che's use case, we'd need @tsmaeder, @svor or @tolusha 's feedbsck on that), or could be unreliable (here it doesn't detect a new pom under a parent project.
So at the very least, we should expose a new import command, users could call explicitly. che-jdt-ls might have something already they could contribute back.
If we do automatic project import, it should be enabled via a preference